src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityQuantumChest.java

```diff
@@ -526,63 +526,78 @@ public ItemStack extractItem(int slot, int amount, boolean simulate) {
 
             // If we had more Items than would fit into the export slot, virtualize the remainder
-            if (amountInsertedIntoExport < stack.getCount()) {
+            if (amountInsertedIntoExport < insertedStack.getCount()) {
                 long amountLeftInChest = itemStack.isEmpty() ? maxStoredItems : maxStoredItems - itemsStoredInside;
-                insertedAmount = (int) Math.min(stack.getCount() - amountInsertedIntoExport, amountLeftInChest);
+                virtualizedAmount = (int) Math.min(insertedStack.getCount() - amountInsertedIntoExport, amountLeftInChest);
 
             }
             // Return early, as we did not virtualize anything, as it all fit into the output slot
             else {
-                return MetaTileEntityQuantumChest.this.exportItems.insertItem(0, stack, simulate);
+                return MetaTileEntityQuantumChest.this.exportItems.insertItem(0, insertedStack, simulate);
             }
 
             ItemStack remainingStack = ItemStack.EMPTY;
 
             // If we are at the maximum that the chest can hold
-            if (stack.getCount() - amountInsertedIntoExport > insertedAmount) {
-                remainingStack = stack.copy();
-                remainingStack.setCount(stack.getCount() - insertedAmount);
+            if (insertedStack.getCount() - amountInsertedIntoExport > virtualizedAmount) {
+                remainingStack = insertedStack.copy();
+                remainingStack.setCount(insertedStack.getCount() - virtualizedAmount);
             }
             if (!simulate) {
-                if (itemStack.isEmpty()) {
-                    MetaTileEntityQuantumChest.this.itemStack = stack.copy();
-                    MetaTileEntityQuantumChest.this.itemsStoredInside = insertedAmount;
-                } else {
-                    MetaTileEntityQuantumChest.this.itemsStoredInside += insertedAmount;
+                if (remainingStack.isEmpty()) {
+                    if (itemStack.isEmpty()) {
+                        ItemStack virtualStack = insertedStack.copy();
+                        virtualStack.setCount(virtualizedAmount);
+                        MetaTileEntityQuantumChest.this.itemStack = virtualStack;
+                        MetaTileEntityQuantumChest.this.itemsStoredInside = virtualizedAmount;
+                    } else {
+                        MetaTileEntityQuantumChest.this.itemsStoredInside += virtualizedAmount;
+                    }
+
+                    if (amountInsertedIntoExport != 0) {
+                        ItemStack insertedStackCopy = insertedStack.copy();
+                        insertedStackCopy.setCount(amountInsertedIntoExport);
+                        MetaTileEntityQuantumChest.this.exportItems.insertItem(0, insertedStackCopy, simulate);
+                    }
+                    return insertedStack;
+                }
+                else {
+                    MetaTileEntityQuantumChest.this.itemsStoredInside += remainingStack.getCount();
```

Comment:
Is this safe to do? What happens when this value exceeds the maximum storage amount in the chest?

Reply:
This will not happen, as `remainingStack` has its count stack size limited by the amount of room left for items to be virtualized